internal/proto/internal.proto

```diff
@@ -67,4 +67,48 @@ message PrContents {
 
   minder.v1.PullRequest pr = 1;
   repeated File files = 2;
-}
\ No newline at end of file
+}
+
+message SelectorProvider {
+  // the name of the provider, e.g. github-app-jakubtestorg
+  string name = 1;
+  // the class of the provider, e.g. github-app
+  string class = 2;
```

Comment:
This message is new in this PR version. It's just a way to avoid repeating what attributes we want to expose for provider.

Reply:
Why is this _both_ in `SelectorRepository`/`SelectorArtifact` _and_ in `SelectorEntity`?  It feels like both copies of the message should have the same contents, so maybe we should have a canonical place?

Reply:
See [the example here](https://github.com/stacklok/minder/pull/3797#discussion_r1675709579). Since the individual selectors can either be targetting a specific entity or any entity (but what is being evaluated is always an specific entity), the canonical place is the entity-specific message (e.g. `SelectorRepository`). What's stored in the generic message (`SelectorEntity`) is just a copy of the common data.

Is that confusing as a concept or would it help to make it crisper in the code in the sense that the provider methods that fill in the selector entity struct just fill in the entity-specific message and the generic attributes are copied by a common function?